CMakeLists.txt

```diff
@@ -269,7 +269,13 @@ execute_process(COMMAND mkdir -p "${CMAKE_BINARY_DIR}/generated")
 
 # We need to link some packages as dynamic/dependent.
 set(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .so)
+if(APPLE)
+  execute_process(COMMAND brew --prefix OUTPUT_VARIABLE BREW_PREFIX)
+  set(OPENSSL_ROOT_DIR "${BREW_PREFIX}/opt/openssl")
```

Comment:
From offline talk with @theopolis, we are going with the second option: Use brew provided openssl and link statically.
brew does install libcrypto.a and libssl.a, so should be just a matter of tweaking CMake and testing it out.